src/SQL_BAIN_OF_MY_EXISTENCE.sql

```diff
@@ -31,10 +31,44 @@ FROM Master
 GROUP BY batter, year; 
 
 -- Beginning of Rolling AVG
-SELECT batter, SUM(Hit)/NULLIF (SUM(atBat),0) as Batting_Avg,
-YEAR (local_date) as year
-FROM Master
-WHERE Master.local_date > ( CURDATE() - INTERVAL 100 DAY )
-GROUP BY batter, Master.local_date; 
+
+-- Rolling Average
+/*DROP TABLE IF EXISTS Rolling_Avg;
+CREATE TABLE Rolling_Avg
+SELECT
+       a.local_date,
+       a.atBat,
+       a.Hit,
+       a.batter,
+       Round( ( SELECT SUM(b.Hit)/NULLIF (SUM(b.atBat),0)
```

Comment:
smart move to avoid the use of subqueries as it would've slowed down your query.